### PR TITLE
feat: children supporting order

### PR DESCRIPTION
**Pull request by chiefcll**
Update renderer for children to support order of nodes.

### CODE REVIEW

File: src/render-drivers/threadx/ThreadXMainNode.ts [hunk 1]
```diff
@@ -165,7 +164,21 @@ export class ThreadXMainNode extends SharedNode implements INode {
...
+    node.parent = this;
+    node.parentId = this?.id ?? 0;
+
+    if (beforeNode) {
+      const index = this.children.indexOf(beforeNode);
+      assertTruthy(
+        index !== -1,
+        'ThreadXMainNode.beforeNode: Node not found in children!',
+      );
+      this.children.splice(index, 0, node);
```
Comment (frank-weindel) [inline src/render-drivers/threadx/ThreadXMainNode.ts:181]: The ThreadX side of this needs some care. To support this, we'd likely need to use messages to communicate parent changes. But then we're getting into talking about having all mutations be communicated my messages in order to ensure everything happens in order 🤔

File: src/render-drivers/main/MainOnlyNode.ts [hunk 2]
```diff
@@ -379,14 +377,28 @@ export class MainOnlyNode extends EventEmitter implements INode {
...
+    node.coreNode.parent = this?.coreNode ?? null;
+
+    if (beforeNode) {
+      const index = this.children.indexOf(beforeNode);
+      assertTruthy(
+        index !== -1,
+        'MainOnlyNode.beforeNode: Node not found in children!',
+      );
+      this.children.splice(index, 0, node);
+    }
```
Comment (elsassph) [inline src/render-drivers/main/MainOnlyNode.ts:399]: Is it missing `else push`?

Comment (elsassph) [reply src/render-drivers/main/MainOnlyNode.ts:399]: Does it?

**Comment by chiefcll** [comment]
Since children are sorted by zIndex this won't work... will need a separate array

**Pull request closed without merging**